src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -92,7 +92,14 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
 
                 using (var process = Process.Start(processStartInfo))
                 {
-                    process.WaitForExit(MsBuildWaitTime);
+                    var finished = process.WaitForExit(MsBuildWaitTime);
+
+                    if (!finished)
+                    {
+                        process.Kill();
```

Comment:
I thought about that. But it can catch other exceptions potentially and hide problems. So I opted to leave it in this change to reduce the risk. It is probably worth doing it when we have more time to react.

The case where this race will happen will be extremely rare, you will need both to time out after 5 minutes and then hit the race condition.

But kill will be called and I don't want to hide anything for now. 

So my assumption is that not catching the exception here would be useful for now